main.c

```diff
@@ -26,38 +26,33 @@ static bool debug = false;
       fprintf(stderr, "DEBUG| " fmt "\n", __VA_ARGS__);                        \
   } while (0)
 
-static void print_vmnet_status(const char *func, vmnet_return_t v) {
+#define INFOF(fmt, ...)     fprintf(stderr, "INFO | " fmt "\n", __VA_ARGS__)
+#define ERROR(msg)          fprintf(stderr, "ERROR| " msg "\n")
+#define ERRORF(fmt, ...)    fprintf(stderr, "ERROR| " fmt "\n", __VA_ARGS__)
+#define ERRORN(name)        ERRORF(name ": %s", strerror(errno))
+
+static const char *vmnet_strerror(vmnet_return_t v) {
   switch (v) {
   case VMNET_SUCCESS:
-    DEBUGF("%s(): vmnet_return_t VMNET_SUCCESS", func);
-    break;
+    return "VMNET_SUCCESS";
   case VMNET_FAILURE:
-    fprintf(stderr, "%s(): vmnet_return_t VMNET_FAILURE\n", func);
-    break;
+    return "VMNET_FAILURE";
   case VMNET_MEM_FAILURE:
-    fprintf(stderr, "%s(): vmnet_return_t VMNET_MEM_FAILURE\n", func);
-    break;
+    return "VMNET_MEM_FAILURE";
   case VMNET_INVALID_ARGUMENT:
-    fprintf(stderr, "%s(): vmnet_return_t VMNET_INVALID_ARGUMENT\n", func);
-    break;
+    return "VMNET_INVALID_ARGUMENT";
   case VMNET_SETUP_INCOMPLETE:
-    fprintf(stderr, "%s(): vmnet_return_t VMNET_SETUP_INCOMPLETE\n", func);
-    break;
+    return "VMNET_SETUP_INCOMPLETE";
   case VMNET_INVALID_ACCESS:
-    fprintf(stderr, "%s(): vmnet_return_t VMNET_INVALID_ACCESS\n", func);
-    break;
+    return "VMNET_INVALID_ACCESS";
   case VMNET_PACKET_TOO_BIG:
-    fprintf(stderr, "%s(): vmnet_return_t VMNET_PACKET_TOO_BIG\n", func);
-    break;
+    return "VMNET_PACKET_TOO_BIG";
   case VMNET_BUFFER_EXHAUSTED:
-    fprintf(stderr, "%s(): vmnet_return_t VMNET_BUFFER_EXHAUSTED\n", func);
-    break;
+    return "VMNET_BUFFER_EXHAUSTED";
   case VMNET_TOO_MANY_PACKETS:
-    fprintf(stderr, "%s(): vmnet_return_t VMNET_TOO_MANY_PACKETS\n", func);
-    break;
+    return "VMNET_TOO_MANY_PACKETS";
   default:
-    fprintf(stderr, "%s(): vmnet_return_t %d\n", func, v);
-    break;
+    return "(unknown status)";
```

Comment:
I did not check the enums values, maybe we can replace this with a simpler table lookup.